# AirBnB NY Locations Data Case Study

In this final project, your task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [19]:
air_bnb = pd.read_csv('../day-1/files/AB_NYC_2019.csv')
air_bnb.head(1)

id                                name  host_id host_name  \
0  2539  Clean & quiet apt home by the park     2787      John   

  neighbourhood_group neighbourhood  latitude  longitude     room_type  price  \
0            Brooklyn    Kensington  40.64749  -73.97237  Private room    149   

   minimum_nights  number_of_reviews last_review  reviews_per_month  \
0               1                  9  2018-10-19               0.21   

   calculated_host_listings_count  availability_365  
0                               6               365

In [92]:
# How many neighborhood groups are available and which shows up the most?

Nbrhood_counts = air_bnb['neighbourhood_group'].value_counts()
Nbrhood_counts

Manhattan        21661
Brooklyn         20104
Queens            5666
Bronx             1091
Staten Island      373
Name: neighbourhood_group, dtype: int64

In [289]:
# Are private rooms the most popular in manhattan?

nbrhood_room_type=air_bnb.groupby(['neighbourhood_group', 'room_type']).size()
nbrhood_room_type


neighbourhood_group  room_type      
Bronx                Entire home/apt      379
                     Private room         652
                     Shared room           60
Brooklyn             Entire home/apt     9559
                     Private room       10132
                     Shared room          413
Manhattan            Entire home/apt    13199
                     Private room        7982
                     Shared room          480
Queens               Entire home/apt     2096
                     Private room        3372
                     Shared room          198
Staten Island        Entire home/apt      176
                     Private room         188
                     Shared room            9
dtype: int64

In [108]:
# Which hosts are the busiest based on their reviews?

reviews = air_bnb[['host_name', 'reviews_per_month']]
sorted_data = reviews.sort_values('reviews_per_month', ascending = False).reset_index()
sorted_data.head()

index host_name  reviews_per_month
0  42075   Row NYC              58.50
1  42076   Row NYC              27.95
2  38870    Louann              20.94
3  27287   Nalicia              19.75
4  28651   Nalicia              17.82

In [262]:
# Which neighorhood group has the highest average price?

air_bnb.groupby('neighbourhood_group').mean()[['price']].sort_values('price', ascending = False).head(1)

price
neighbourhood_group            
Manhattan            196.875814

In [273]:
# Which neighborhood group has the highest total price?
air_bnb[['neighbourhood_group','price']].groupby('neighbourhood_group').sum(['price']).sort_values('price', ascending = False).head(1)

price
neighbourhood_group         
Manhattan            4264527

In [239]:
# Which top 5 hosts have the highest total price?
air_bnb[['host_id','price']].groupby(by="host_id").sum(['price']).sort_values('price', ascending = False).head()

price
host_id         
219517861  82795
107434423  70331
156158778  37097
205031545  35294
30283594   33581

In [296]:
# Who currently has no (zero) availability with a review count of 100 or more?

no_availability=air_bnb[air_bnb['availability_365']==0]
no_availability
hundred_reviews_plus_no_avail=no_availability[no_availability['number_of_reviews']>=100].sort_values('number_of_reviews', ascending = False)
hundred_reviews_plus_no_avail

id                                               name    host_id  \
471      166172                    LG Private Room/Family Friendly     792159   
9974    7670562       JFK 10 & LGA 15 MINUTES A/C  PRIVATE BEDROOM   22959695   
9976    7675781  JFK 10 & LGA 15 MINUTES  AWAY  A/C PRIVATE BED...   22959695   
22104  17779504  Relaxing Quiet Room Just 6 Mins from JFK Airport!  121391142   
5876    4282178                     A Superhost SALE! DELUXE Room!   21641206   
...         ...                                                ...        ...   
19377  15462208       Spacious 2 Bedroom Apartment by Central Park   26073602   
12375   9544914  King Bed, Private Room in Crown Heights, Comedian   22423049   
16190  13065356      "Quaint-Essential" Living in the West Village   42399786   
19459  15523119                   3 bedroom, second floor and cozy   96148809   
18230  14283546         Adorable Crown Heights Home Away From Home   84141923   

             host_name neighbourhood_group        neighbourhood  latitude  \
471              Wanda            Brooklyn             Bushwick  40.70283   
9974   Gurpreet  Singh              Queens        Richmond Hill  40.69604   
9976   Gurpreet  Singh              Queens        Richmond Hill  40.69463   
22104          Deloris              Queens  Springfield Gardens  40.66535   
5876          Veronica            Brooklyn   Bedford-Stuyvesant  40.69724   
...                ...                 ...                  ...       ...   
19377             Anna           Manhattan          East Harlem  40.79575   
12375          Abraham            Brooklyn        Crown Heights  40.67157   
16190          Braydon           Manhattan         West Village  40.73564   
19459          Raymond            Brooklyn   Bedford-Stuyvesant  40.68227   
18230          Marisha            Brooklyn        Crown Heights  40.66902   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
471    -73.92131     Private room     60               3                480   
9974   -73.82449     Private room     50               1                424   
9976   -73.82610     Private room     50               1                408   
22104  -73.76367     Private room     65               1                368   
5876   -73.94914     Private room     53               1                351   
...          ...              ...    ...             ...                ...   
19377  -73.94817  Entire home/apt    145               2                101   
12375  -73.94117     Private room     36               2                100   
16190  -73.99823  Entire home/apt    145               3                100   
19459  -73.93479  Entire home/apt    150               2                100   
18230  -73.93771     Private room     50               3                100   

      last_review  reviews_per_month  calculated_host_listings_count  \
471    2019-07-07               6.70                               1   
9974   2018-11-25               8.86                               5   
9976   2018-11-24               8.56                               5   
22104  2019-06-21              13.24                               2   
5876   2019-06-24               6.09                               4   
...           ...                ...                             ...   
19377  2019-01-12               3.07                               1   
12375  2017-07-31               2.28                               1   
16190  2018-09-15               2.62                               1   
19459  2019-07-05               3.26                               1   
18230  2019-02-19               2.81                               2   

       availability_365  
471                   0  
9974                  0  
9976                  0  
22104                 0  
5876                  0  
...                 ...  
19377                 0  
12375                 0  
16190                 0  
19459                 0  
18230          

In [279]:
# What host has the highest total of prices and where are they located?

air_bnb[['host_name','price', 'neighbourhood_group']].groupby('host_name').sum(['price']).sort_values('price', ascending = False).head()                                                      
                                                              

price
host_name          
Sonder (NYC)  82795
Blueground    70331
Michael       66895
David         65844
Alex          52563

In [272]:
# When did Danielle from Queens last receive a review?

danielle_queens = air_bnb[air_bnb['neighbourhood_group']=='Queens']
danielle = air_bnb_queens[danielle_queens['host_name']=='Danielle']
danielle['last_review'].sort_values()

27021    2018-01-02
33861    2019-06-20
7086     2019-07-03
20403    2019-07-06
22068    2019-07-06
21517    2019-07-07
22469    2019-07-08
16349           NaN
Name: last_review, dtype: object

## Further Questions

1. Which host has the most listings?

In [261]:
air_bnb[['host_id', 'id']].groupby(['host_id']).count().sort_values('id', ascending = False).head(1)

id
host_id       
219517861  327

2. How many listings have completely open availability?

In [250]:
air_bnb[['availability_365']][air_bnb['availability_365']==365].count()

availability_365    1295
dtype: int64

3. What room_types have the highest review numbers?

In [253]:
air_bnb[['room_type', 'number_of_reviews']].groupby(['room_type']).sum().sort_values('number_of_reviews', ascending = False)

number_of_reviews
room_type                         
Entire home/apt             580403
Private room                538346
Shared room                  19256

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here. 


1 --- # How many neighborhood groups are available and which shows up the most?

5 Neighborhood Groups (Manhatten, Brooklyn, Queens, Bronx, Staten Island) - Manhatten shows up the most.

2 --- # Are private rooms the most popular in manhattan?

Private Rooms are the most popular room type in all other neighborgood groups OTHER THAN Manhatten. In Manhatten, the Entire Home/Apt are the most popular room type.

3 --- # Which hosts are the busiest based on their reviews?

Top 5 Hosts (RPM):
Row NYC - (58.50 RPM)
Row NYC - (27.95 RPM) 
Louann - (20.94 RPM)
Nalicia - (19.75 RPM) 
Nalicia - (17.82 RPM)

4 --- # Which neighorhood group has the highest average price?

Manhattan (196.86)

5 --- # Which neighborhood group has the highest total price?

Manhatten (4,264,527)

6 --- # Which top 5 hosts have the highest total price?

Top 5 Hosts (Total Price):
host_id: 219517861 with a total price of 82,795
host_id: 107434423 with a total price of 70,331
host_id: 156158778 with a total price of 37,097
host_id: 205031545 with a total price of 35,294
host_id: 30283594  with a total price of 33,581


7 --- # Who currently has no (zero) availability with a review count of 100 or more?

162 listings have no availability and over 100 reviews. For some added information, it looks like the most reviewed property with no availability comes from Wanda in Brooklyn with 480 reviews!

8 --- # What host has the highest total of prices and where are they located?

Sonders (NYC) has the highest total prices and is located in Manhatten.

9 --- # When did Danielle from Queens last receive a review?

Last recorded review for Danielle in Queens was 2019-07-08.


Final 3 Questions:

1 --- Host 219517861 has the most listings.

2 --- 1,295 listings have open availability.

3 --- Entire Homes/Apts have the highest reviews numbers (total amount of reviews). 